src/pool.rs

```diff
@@ -375,8 +375,8 @@ impl Pool {
                                          }
                                      };
                                      let old =
-                                         shared.finished.fetch_add(1, atomic::Ordering::SeqCst);
-                                     let sent = shared.sent.load(atomic::Ordering::SeqCst);
+                                         shared.finished.fetch_add(1, atomic::Ordering::AcqRel);
```

Comment:
*grins* I'd love to give you a more intelligent answer, but the reality is that it's just been over the last couple of months that I've been getting into lock free programming, and at the time of the PR I wasn't feeling competent to judge use of relaxed orderings - where I felt on much more confident with Acquire/Release vs sequential consistency.